build.gradle.kts

```diff
@@ -13,6 +13,9 @@ plugins {
 java {
     sourceCompatibility = JavaVersion.VERSION_1_8
     targetCompatibility = JavaVersion.VERSION_1_8
+    toolchain {
+        languageVersion = JavaLanguageVersion.of(11)
```

Comment:
You bumped the Java level from 8n to 11, we can use toolchain here, and remove `sourceCompatibility` and `targetCompatibility`.

Reply:
Yes the story is more complicated here...

You guys are already using Java 11 in your CI run. So I wanted to keep you on 11. See
https://github.com/spotbugs/spotbugs-gradle-plugin/blob/17eccf7c58573576dee682b4669f6f76d7180d26/.github/workflows/gradle.yml#L24

But in addition to that, we are not able to downgrade to 8 since errorprone only supports Java 11+. We also cannot upgrade to 17 since this would break some testcases (though we could just adapt those tests). So JDK 11 is really our only choice.

If we would adapt target/sourceCompatibility, this could break some users of your Gradle Plugin since they may still be running Gradle on JDK 8.

Since I didn't want to introduce a breaking change and also wanted to be as non-invasive as possible, I came to the current configuration with source/target-Compat at JDK 8 and toolchain at JDK 11.

Does that make sense to you? :)

Reply:
> If we would adapt target/sourceCompatibility, this could break some users of your Gradle Plugin since they may still be running Gradle on JDK 8.

That's the point, after this, it requires all consumers running on Java 11 above, we should keep the compatibility of Java 8.

Reply:
I still fear we might have a misunderstanding here.

Previously on your CI, you also used JDK 11 to run Gradle. This also meant that JDK 11 was used to compile the plugin. But with source/targetCompat, we told the compiler to emit Java 8 classes.

Now, we basically force Gradle to compile and test with JDK 11, no matter on which JVM Gradle was started. This won't change anything on your CI, since you correctly used JDK 11 anyways. But on dev machines, this will lead to everyone executing compile and test with the same JDK, no matter what was previously installed on the system. This increases reproducability. But even with this change, we still emit Java 8 class files due to source/target compat. I checked this by hand.

Reply:
You are right. I viewed the output jar, it's still on Java 8.

```sh
./gradlew assemble
cd ./build/libs  
jar xf ./spotbugs-gradle-plugin.jar
javap -v com/github/spotbugs/snom/Confidence.class  | grep major
  major version: 52
```

Reply:
Soo....I guess we can consider this comment resolved? :) Or do you have further remarks?

Reply:
Yeah, seems we can consider using Java 17?

Reply:
We could consider it, yeah. Currently, JDK 17 breaks `ExtensionFunctionalTest#can use toolVersion to get the SpotBugs version`. We'd have to fix this somehow.

I could look into this. Maybe in a separate PR? Or should I do it in this PR?

Reply:
Yeah, it could be in a follow-up.